baremetal/metal3data_manager.go

```diff
@@ -610,11 +709,14 @@ func (m *DataManager) getAddressFromPool(ctx context.Context, poolName string,
 		gateway = *ipAddress.Spec.Gateway
 	}
 
-	// get address, gateway and prefixes
+	dnsServers := ipAddress.Spec.DNSServers
+
+	// get address, gateway, prefix and dns servers
```

Comment:
remove '// get address, gateway, prefix and dns servers'
we can what it returns